x/interchainqueries/types/tx.go

```diff
@@ -159,6 +180,10 @@ func (msg MsgUpdateInterchainQueryRequest) ValidateBasic() error {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "one of new_keys or new_update_period should be set")
 	}
 
+	if uint64(len(msg.NewKeys)) > MaxKVQueryKeysCount {
+		return sdkerrors.Wrapf(ErrTooManyKVQueryKeys, "keys count cannot be more than %d", MaxKVQueryKeysCount)
+	}
```

Comment:
we have such a thorough validation on query registration, but on query update we barely validate it, only the length. I believe we should have the same validation on update as we have on registration, e.g. move the algorithm to a standalone func and reuse in both places

Reply:
moved

Reply:
No no, I meant, we have the following checks on query registration: keys number is not too much, not zero, each key payload. But on update we only check whether there are not too many keys, which means I can possibly pass keys with invalid payload.